napari/components/_layer_slicer.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from concurrent.futures import Executor, Future, ThreadPoolExecutor
+from contextlib import contextmanager
+from typing import TYPE_CHECKING, Dict, Iterable, Tuple
+
+from napari.layers import Layer
+from napari.utils.events.event import EmitterGroup, Event
+
+from . import Dims
+
+if TYPE_CHECKING:
+    from napari.layers.base.base import _LayerSliceRequest, _LayerSliceResponse
+
+LOGGER = logging.getLogger("napari.components._layer_slicer")
+
+
+class _LayerSlicer:
+    def __init__(self):
+        self.events = EmitterGroup(source=self, ready=Event)
+        self._executor: Executor = ThreadPoolExecutor(max_workers=1)
+        self._layers_to_task: Dict[Tuple[Layer], Future] = {}
+        self._force_sync = False
```

Comment:
Consider starting without these. We can add cancellation and force-sync behavior later. I think we should start with just slice execution.